api/client/client.go

```diff
@@ -16,13 +15,14 @@ type Client struct {
 }
 
 // NewClient starts the client
-func NewClient(maddr ma.Multiaddr) (*Client, error) {
-	addr, err := util.TCPAddrFromMultiAddr(maddr)
-	if err != nil {
-		return nil, err
+func NewClient(target string, creds credentials.TransportCredentials) (*Client, error) {
+	var opts []grpc.DialOption
+	if creds != nil {
+		opts = append(opts, grpc.WithTransportCredentials(creds))
+	} else {
+		opts = append(opts, grpc.WithInsecure())
 	}
-	// ToDo: Support secure connection
-	conn, err := grpc.Dial(addr, grpc.WithInsecure())
+	conn, err := grpc.Dial(target, opts...)
```

Comment:
Moved the grpc connection code to the way @sanderpick started doing it in `textile`.

Reply:
Cool! In the threads client(s), I started just passing raw opts to give the user max flexibility... not sure if needed here.

Reply:
Makes sense, will do that here